### ISSUE TITLE
Lmms-1.3.0-alpha (appimage from website) almost no LV2 Plugins

### ISSUE THREAD

**Issue by Daniele71** [LMMS/lmms#5832]
Hello, I know that LV2 support isn't finished, so I'm not sure if this is a bug or not.. anyway.. Plugin manager only shows 10 plugins, all from eq10q family. Also eq10q is loaded with a generic gui, not very useful..

Opensuse TW.

Thanks,
Daniele.

**Comment by zonkmachine** [comment]
Yes, this is correct. LV2 is under development and is a modular system so it will only show the most rudimentary support on the keyboard side. You can try the latest test build and see if some more plugins will show up. https://github.com/LMMS/lmms/pull/5761#issuecomment-720146140 (follow link and click the robot).

**Comment by JohannesLorenz** [comment]
@Daniele71 any news? What if you install other Lv2 plugin collections (CAPS, CALF, SWH, ...)? Do they show up? On Linux, you can start `LMMS_LV2_DEBUG=1 /path/to/lmms` to get explanations why the plugins are rejected.

**Comment by Daniele71** [comment]
> @Daniele71 any news? What if you install other Lv2 plugin collections (CAPS, CALF, SWH, ...)? Do they show up? On Linux, you can start `LMMS_LV2_DEBUG=1 /path/to/lmms` to get explanations why the plugins are rejected.

Hi, is there a new Nightly build to test ? I'm not using Lmms right now but i can do quick test...

**Comment by JohannesLorenz** [comment]
> Hi, is there a new Nightly build to test ? I'm not using Lmms right now but i can do quick test...

@Daniele71 Just take it from any recent PR.

**Comment by Daniele71** [comment]
> > Hi, is there a new Nightly build to test ? I'm not using Lmms right now but i can do quick test...
> 
> @Daniele71 Just take it from any recent PR.
A link please ?! Why nightly build is not update in the download section ?

**Comment by JohannesLorenz** [comment]
@Daniele71 Check the LmmsBot's post in any PR, e.g. : https://github.com/LMMS/lmms/pull/6331

**Comment by Daniele71** [comment]
> @Daniele71 Check the LmmsBot's post in any PR, e.g. : #6331

Thanks but appimage link lead to a blank page with: {"message":"unauthorized"}

**Comment by JohannesLorenz** [comment]
What does it output for `LMMS_LV2_DEBUG=1` ?

**Comment by Daniele71** [comment]
> LMMS_LV2_DEBUG=1
Few errors about a Harrinson plugin.
Lv2 plugin SUMMARY: 18 of 50  loaded in 15 msecs.

For Carla lv2 plugins are 805

**Comment by JohannesLorenz** [comment]
If it only finds 50 out of 805, then there is something strange. Does Carla have an additional path for the Lv2 Plugins? Where are these 50 plugins located, and where are the 805-50 other ones located?

**Comment by Daniele71** [comment]
> 
few plugins in ~/.lv2 (but not 50, 3-4 ) then /usr/lib64/lv2

**Comment by JohannesLorenz** [comment]
Where are the 50 located and where are the 805-50 located? In the same directory?

**Comment by Daniele71** [comment]
> Where are the 50 located and where are the 805-50 located? In the same directory?

The 50 are a mistery for me. All plugins are in /usr/lib64/lv2 except 3 that are in ~/.lv2

**Comment by JohannesLorenz** [comment]
@Daniele71 If I remember correctly, you can start `LV2_PATH=/usr/lib64/lv2 /path/to/appimage`. Does it then show you the 527 plugins?

**Comment by Daniele71** [comment]
> @Daniele71 If I remember correctly, you can start `LV2_PATH=/usr/lib64/lv2 /path/to/appimage`. Does it then show you the 527 plugins?
Lv2 plugin SUMMARY: 587 of 772  loaded in 2692 msecs.
For details about not loaded plugins, please set
  environment variable "LMMS_LV2_DEBUG" to nonempty.
Lv2 Plugins blacklisted: 10 of 772 

But only 31 plugins are available inside Lmms.

**Comment by JohannesLorenz** [comment]
1. Does this mean that setting the `LV2_PATH` increase the number of "loaded" plugins?
2. If it loaded 587 plugins, they *should* be available... How many are in the instrument side bar, and how many (approximately) have the type "Lv2" in the window where you choose an effect to add?

**Comment by Daniele71** [comment]
> 1. Does this mean that setting the `LV2_PATH` increase the number of "loaded" plugins?
> 2. If it loaded 587 plugins, they _should_ be available... How many are in the instrument side bar, and how many (approximately) have the type "Lv2" in the window where you choose an effect to add?

31 in the sidebar all marked as Lv2.

**Comment by JohannesLorenz** [comment]
Please answer to both questions. Question 1 is not answered, and question 2 is incomplete: do you really see 0 Lv2 Effects?

**Comment by JohannesLorenz** [comment]
@Daniele71 I you don't reply within the next days, I'll close this as invalid. It looks like you just oversaw that most plugins are effects.

**Comment by Daniele71** [comment]
> @Daniele71 I you don't reply within the next days, I'll close this as invalid. It looks like you just oversaw that most plugins are effects.
Recap, default:
Lv2 plugin SUMMARY: 18 of 50  loaded in 15 msecs.
2 plugin in sidebar and 14 as FX

With LV2_PATH=/usr/lib64/lv2
Lv2 plugin SUMMARY: 588 of 773  loaded in 2398 msecs.
34 under sidebar and a few hundreds as FX


In the sidebar some instruments  plugin are missing. I've both Surge an Surge XT installed but only Surge appears. Also ZynAddSubFx is missing.

**Comment by PhysSong** [comment]
It looks like lilv in the AppImage(which from Ubuntu) doesn't look at `/usr/lib64/lv2`. The directory list is set at compile time as `~/.lv2:/usr/lib/$(DEB_HOST_MULTIARCH)/lv2:/usr/lib/lv2:/usr/local/lib/lv2`(see `rules` file in http://archive.ubuntu.com/ubuntu/pool/universe/l/lilv/lilv_0.24.2~dfsg0-1.debian.tar.xz).
To work around this issue, we have to ship a manually configured or patched version of lilv for AppImage builds.

**Comment by anytizer** [comment]
First line of Debug message in **Windows**: `Lv2 plugin SUMMARY: 0 of 0 loaded in 0 msecs.`
